lib/rules/lines-around-comment.js

```diff
@@ -231,9 +231,15 @@ module.exports = {
             const parent = getParentNodeOfToken(token);
 
             if (parent && isParentNodeType(parent, nodeType)) {
-                const parentStartNodeOrToken = parent.type === "StaticBlock"
-                    ? sourceCode.getFirstToken(parent, { skip: 1 }) // opening brace of the static block
-                    : parent;
+                let parentStartNodeOrToken = parent;
+
+                if (parent.type === "StaticBlock") {
+                    parentStartNodeOrToken = sourceCode.getFirstToken(parent, { skip: 1 }); // opening brace of the static block
+                } else if (parent.type === "SwitchStatement") {
+                    parentStartNodeOrToken = sourceCode.getFirstToken(parent, {
+                        filter: t => t.type === "Punctuator" && t.value === "{"
+                    }); // opening brace of the switch statement
```

Comment:
```suggestion
                    parentStartNodeOrToken = sourceCode.getTokenAfter(parent.discriminant, {
                        filter: astUtils.isOpeningBraceToken
                    }); // opening brace of the switch statement
```

Some `{` can appear before the `{` we are looking for:

```js
/* eslint lines-around-comment: ["error", {
    allowBlockStart: true,
    beforeLineComment: true
}] */

switch (
    function(){}()
)
{
    // this comment is a false positive
    case foo:
        break;
}

```

Reply:
Thanks! Fixed 👍🏻